bazel_tools/scala_version.bzl

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+def _impl(ctx):
+    # Generates an empty BUILD file, because we do not need to build anything.
+    ctx.file(
+        "BUILD",
+        content = """exports_files(["index.bzl"])""",
+        executable = False,
+    )
+
+    version = ctx.os.environ.get("DAML_SCALA_VERSION", default = "2.12.12")
+    suffix = version.replace(".", "_")
+
+    last_dot = version.rfind(".")
+    major = version[:version.rfind(".")]
+    major_suffix = major.replace(".", "_")
+    ctx.file(
+        "index.bzl",
+        content =
+            """
+scala_version = "{version}"
+scala_major_version = "{major}"
+scala_version_suffix = "{suffix}"
+scala_major_version_suffix = "{major_suffix}"
+""".format(
+                version = version,
+                major = major,
+                suffix = suffix,
+                major_suffix = major_suffix,
+            ),
+        executable = False,
+    )
+
+scala_version = repository_rule(
+    environ = ["DAML_SCALA_VERSION"],
+    implementation = _impl,
+    attrs = {},
+)
```

Comment:
This is a configure type rule so it should also set [`configure = True`](https://docs.bazel.build/versions/3.3.0/skylark/lib/globals.html#repository_rule.configure), see [here](https://docs.bazel.build/versions/master/skylark/repository_rules.html#forcing-refetch-of-external-repositories) for the effect.

Reply:
I don’t quite understand this. I thought `environ` captures the cases where it needs to rebuild?

Reply:
That seems to only matter if we explicitly run `bazel sync --configure`. am I reading this wrong?

Reply:
Yes, @garyverhaegen-da's matches my understanding of it. We don't really use this, but it doesn't hurt to follow the recommended pattern.

Reply:
I still don’t understand it. The way I understand the docs this is for things that depend on state not captured by the rule. But here we only depend on the environment variable which is captured.

Reply:
Merged  to avoid conflicts but very happy to revisit this.

Reply:
Hmm, I tested it and Bazel refetches this rule with `bazel sync --configure` without `configure=True` even if the envvar is not changed.